src/main/cpp/src/parse_uri.cu

```diff
@@ -887,24 +956,43 @@ std::unique_ptr<column> parse_uri_to_protocol(strings_column_view const& input,
                                               rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
-  return detail::parse_uri(input, detail::URI_chunks::PROTOCOL, stream, mr);
+  return detail::parse_uri(input, detail::URI_chunks::PROTOCOL, std::nullopt, stream, mr);
 }
 
 std::unique_ptr<column> parse_uri_to_host(strings_column_view const& input,
                                           rmm::cuda_stream_view stream,
                                           rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
-  return detail::parse_uri(input, detail::URI_chunks::HOST, stream, mr);
+  return detail::parse_uri(input, detail::URI_chunks::HOST, std::nullopt, stream, mr);
 }
 
 std::unique_ptr<column> parse_uri_to_query(strings_column_view const& input,
                                            rmm::cuda_stream_view stream,
                                            rmm::mr::device_memory_resource* mr)
 {
   CUDF_FUNC_RANGE();
-  return detail::parse_uri(
-    input, detail::URI_chunks::QUERY, stream, rmm::mr::get_current_device_resource());
+  return detail::parse_uri(input, detail::URI_chunks::QUERY, std::nullopt, stream, mr);
+}
+
+std::unique_ptr<cudf::column> parse_uri_to_query(cudf::strings_column_view const& input,
+                                                 std::string const query_match,
+                                                 rmm::cuda_stream_view stream,
+                                                 rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+
+  // build string_column_view from incoming query_match string
+  auto d_scalar        = make_string_scalar(query_match, stream);
+  auto d_string_scalar = static_cast<cudf::string_scalar*>(d_scalar.release());
+  auto x = thrust::pair<char const*, size_type>{d_string_scalar->data(), d_string_scalar->size()};
+  auto y = std::vector{x};
+  auto z = host_span<thrust::pair<char const*, size_type>>(y);
+  auto d_vector = cudf::detail::make_device_uvector_async<thrust::pair<char const*, size_type>>(
+    z, stream, rmm::mr::get_current_device_resource());
+  auto col = make_strings_column(d_vector, stream, rmm::mr::get_current_device_resource());
```

Comment:
I think all of this could be replaced with `make_column_from_scalar`

Reply:
Thank you, I was looking for this and didn't find it. Much, much cleaner.